src/Illuminate/Queue/Worker.php

```diff
@@ -538,19 +573,23 @@ protected function markJobAsFailedIfWillExceedMaxAttempts($connectionName, $job,
      */
     protected function markJobAsFailedIfWillExceedMaxExceptions($connectionName, $job, Throwable $e)
     {
-        if (! $this->cache || is_null($uuid = $job->uuid()) ||
+        if (! $this->cache || is_null($job->uuid()) ||
             is_null($maxExceptions = $job->maxExceptions())) {
             return;
         }
 
-        if (! $this->cache->get('job-exceptions:'.$uuid)) {
-            $this->cache->put('job-exceptions:'.$uuid, 0, Carbon::now()->addDay());
-        }
+        $jobExceptionsKey = $this->getJobTotalExceptionsKey($job);
 
-        if ($maxExceptions <= $this->cache->increment('job-exceptions:'.$uuid)) {
```

Comment:
A race condition has been introduced here. Previously, the `increment` operation here was atomic for some cache systems (Redis, Database, etc.)... but now it has been changed to read, perform some PHP code, and then increment in certain conditions, meaning there could be a race condition between the read and the increment that could lead to unexpected flow.

Reply:
@taylorotwell Thanks for your review! I've changed the way it increases and compares, to avoid the race condition. Could you please take a look at this?